example-application/api.js

```diff
@@ -106,6 +99,20 @@ const defineRoutes = (expressApp) => {
     res.status(204).end();
   });
 
+  expressApp.use((req, res, next) => {
+    const start = Date.now();
+
+    res.on('finish', async () => {
+      const duration = Date.now() - start;
+
+      if (duration > 1 * 3000) {
+        await metricsExporter.fireMetric('too slow response');
```

Comment:
Two monitoring remarks:

- This logic should happen in the monitoring side, here we just throw all the data to it (no IF), and the decision what is bad/good is made on there based on data you don't have here

- In most of the monitoring system that I'm familiar with the metric name convention is 'too_slow_response' (Prometheus, Datadog)

Reply:
Right.
I removed this